LegionBoard: Respect archived courses/teachers

AltNico opened the pull request:
Courses and teachers that have the archived field set to "true" should not be shown in the selection. This does not affect generating changes.

It is done in Eye too: https://gitlab.com/legionboard/eye/blob/master/src/js/changes/show.js#L274

See also: https://gitlab.com/legionboard/heart/blob/master/doc/teachers/update.md

johan12345 commented on parser/src/main/java/me/vertretungsplan/parser/LegionBoardParser.java:
```diff
@@ -272,7 +272,9 @@ private Substitution getSubstitution(JSONObject change, HashMap<String, String>
 		for (int i = 0; i < courses.length(); i++) {
 			final JSONObject course = courses.getJSONObject(i);
-			classes.add(course.getString("name"));
+			if (course.getString("archived").equals("false")) {
```
Isn't this a boolean value in the JSON? (-> if yes, use `getBoolean` instead of `getString`).

johan12345 replied:
> Maybe I should fix this in Heart...

Yes, sounds reasonable (but then needs to be fixed in the other clients, too).

johan12345 commented on parser/src/main/java/me/vertretungsplan/parser/LegionBoardParser.java:
```diff
@@ -287,7 +289,9 @@ private Substitution getSubstitution(JSONObject change, HashMap<String, String>
 		for (int i = 0; i < jsonTeachers.length(); i++) {
 			final JSONObject teacher = jsonTeachers.getJSONObject(i);
-			teachers.add(teacher.getString("name"));
+			if (teacher.getString("archived").equals("false")) {
```
see above

AltNico commented on parser/src/main/java/me/vertretungsplan/parser/LegionBoardParser.java:
```diff
@@ -272,7 +272,9 @@ private Substitution getSubstitution(JSONObject change, HashMap<String, String>
 		for (int i = 0; i < courses.length(); i++) {
 			final JSONObject course = courses.getJSONObject(i);
-			classes.add(course.getString("name"));
+			if (course.getString("archived").equals("false")) {
```
No, it is indeed served as a string:

https://gitlab.com/legionboard/heart/blob/master/src/lib/resources/teachers.php#L35

Maybe I should fix this in Heart...

AltNico replied:
Reported: https://gitlab.com/legionboard/heart/issues/17

AltNico commented:
I just realized I linked to the wrong doc in the commit message.
https://gitlab.com/legionboard/heart/blob/master/doc/changes/list.md#success -> https://gitlab.com/legionboard/heart/blob/master/doc/teachers/update.md

AltNico committed:
LB: respect archived courses/teachers

Courses and teachers that have the archived field set to "true" should not be shown in the selection. This does not affect generating changes.

It is done in Eye too: https://gitlab.com/legionboard/eye/blob/master/src/js/changes/show.js#L274

See also: https://gitlab.com/legionboard/heart/blob/master/doc/teachers/update.md

AltNico commented:
Done.
See https://gitlab.com/legionboard/heart/commit/632a6a7514b227517776b764b673dfb525dc3731

johan12345 commented:
Thanks! I merged both your PRs, released a new version and updated it on the server.
